src/main/java/nl/ordina/jobcrawler/service/SkillService.java

```diff
@@ -1,130 +1,90 @@
-                newSkills.add(skill);
-            }
-        }
-        return newSkills;
+    /**
+     * Returns all skills in the database.
+     *
+     * @return All skills in the database.
+     */
+    @Override
+    public List<Skill> findAll() {
+        return skillRepository.findAll();
     }
 
-
-    //******** Deleting ********//
-    // Deletes from the skill table if no relations are found in the linking table
-    public void deleteSkillsIfNoRelations(Set<Skill> skillsToDelete) {
-        for (Skill skill : skillsToDelete) {
-            int relations = skillRepository.countRelationsById(skill.getId());  //counts the entries in the relation table
-            System.out.println("** Skill " + skill.getName() + " has " + relations + " relation(s)");
-            if (relations == 0) {
-                System.out.println("** Deleting skill " + skill.getName());
-                skillRepository.deleteById(skill.getId()); // deletes the skill from the skill table
-            }
-        }
+    /**
+     * Updates the a skill, identified by its id.
+     *
+     * @param newSkill The skill with the values to be updated.
+     * @return True if the update succeeded, otherwise false.
+     */
+    @Override
+    public Skill update(UUID id, Skill newSkill) {
+
+        return skillRepository.findById(id)
+                .map(oldSkill -> {
+                    oldSkill.setName(newSkill.getName());
+                    return skillRepository.save(oldSkill);
+                }).orElseThrow(() -> new SkillNotFoundException(id));
     }
 
 
-    //******** Adding ********//
-    // takes skills and adds it to a vacancy //
-    public void addSkillsToVacancy(Set<Skill> skills, Vacancy vacancy) { ;
-        skills = linkToExistingSkills(skills);
-        vacancy.addSkills(skills);
+    /**
+     * Saves the specified skill to the database.
+     *
+     * @param skill The skill to save to the database.
+     * @return The saved skill.
+     */
+    @Override
+    public Skill save(Skill skill) {
+        return skillRepository.save(skill);
     }
 
-
-    //******** Deleting ********//
-    // takes skills and removes them from a vacancy //
-    public void removeSkillsFromVacancy(Set<Skill> skills, Vacancy vacancy) {
-        for (Skill skill : skills) {
-            // remove the relationships, this can also be done using the vacancyRepository
-            skillRepository.removeRelationsById(skill.getId(), vacancy.getId());
+    /**
+     * Deletes the skill with the specified id.
+     *
+     * @param id The id of the skill to delete.
+     * @return True if the operation was successful, false otherwise.
+     */
+    @Override
+    public boolean delete(UUID id) {
+        try {
+            skillRepository.deleteById(id);
+            return true;
+        } catch (JDBCConnectionException e) {
+            return false;
         }
-        vacancy.removeSkills(skills);
-        deleteSkillsIfNoRelations(skills);
     }
```

Comment:
This exception is not could but is thrown when the connection to the database fails. I am not sure what exception to be put here so that if service could not execute a delete it would return a false.